pkg/controller/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,117 @@
+package bootstrap
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/pkg/controller/render"
+	"github.com/openshift/machine-config-operator/pkg/controller/template"
+	"github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/scheme"
+)
+
+// Bootstrap defines boostrap mode for Machine Config Controller
+type Bootstrap struct {
+	// dir used by template controller to render internal machineconfigs.
+	templatesDir string
+	// dir used to read pools and user defined machineconfigs.
+	manifestDir string
+}
+
+// New returns controller for bootstrap
+func New(templatesDir, manifestDir string) *Bootstrap {
+	return &Bootstrap{
+		templatesDir: templatesDir,
+		manifestDir:  manifestDir,
+	}
+}
+
+// Run runs boostrap for Machine Config Controller
+// It writes all the assets to destDir
+func (b *Bootstrap) Run(destDir string) error {
+	infos, err := ioutil.ReadDir(b.manifestDir)
+	if err != nil {
+		return err
+	}
+
+	var cconfig *v1.ControllerConfig
+	var pools []*v1.MachineConfigPool
+	var configs []*v1.MachineConfig
+	for _, info := range infos {
+		if info.IsDir() {
+			continue
+		}
+
+		path := filepath.Join(b.manifestDir, info.Name())
+		raw, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		obji, err := runtime.Decode(scheme.Codecs.UniversalDecoder(v1.SchemeGroupVersion), raw)
+		if err != nil {
+			glog.V(4).Infof("skipping path because of error: %v", err)
```

Comment:
Maybe we want to print out the error by default? why we don’t care about the error? if the manifests are not unmarshaled correctly, that means we are not getting the expected result right?

Reply:
the decoder only decodes `machineconfiguration.openshift.io/v1` types.

If there are other manifests in directory like we have in `/opt/tectonic/manifests`; they will not be decoded, and we don't care.

Reply:
ok